components/datetime/src/fieldsets.rs

```diff
@@ -314,56 +314,18 @@ macro_rules! impl_zone_combo_helpers {
         $enum:ident
     ) => {
         impl $type {
-            /// Associates this field set with a long specific non-location format time zone, as in
-            /// “Pacific Daylight Time”.
-            #[inline]
-            pub fn with_zone_specific_long(self) -> Combo<Self, Z> {
-                Combo::new(self, Z::new())
-            }
-            /// Associates this field set with a short specific non-location format time zone, as in
-            /// “PDT”.
-            #[inline]
-            pub fn with_zone_specific(self) -> Combo<Self, Zs> {
-                Combo::new(self, Zs::new())
-            }
-            /// Associates this field set with a long offset format time zone, as in
-            /// “GMT−08:00”.
-            #[inline]
-            pub fn with_zone_offset_long(self) -> Combo<Self, O> {
-                Combo::new(self, O::new())
-            }
-            /// Associates this field set with a short offset format time zone, as in
-            /// “GMT−8”.
-            #[inline]
-            pub fn with_zone_offset(self) -> Combo<Self, Os> {
-                Combo::new(self, Os::new())
-            }
-            /// Associates this field set with a long generic non-location format time zone, as in
-            /// “Pacific Time”.
-            #[inline]
-            pub fn with_zone_generic_long(self) -> Combo<Self, V> {
-                Combo::new(self, V::new())
-            }
-            /// Associates this field set with a short generic non-location format time zone, as in
-            /// “PT”.
-            #[inline]
-            pub fn with_zone_generic(self) -> Combo<Self, Vs> {
-                Combo::new(self, Vs::new())
-            }
-            /// Associates this field set with a location format time zone, as in
-            /// “Los Angeles time”.
-            #[inline]
-            pub fn with_zone_location(self) -> Combo<Self, L> {
-                Combo::new(self, L::new())
+            /// Associates this field set with a time zone format.
+            pub fn with_zone<Z: ZoneMarkers>(self, zone: Z) -> Combo<Self, Z> {
```

Comment:
Praise: This new signature for `with_zone` is nice.

Reply:
Well, if you look at the `timezone_picker` example, that requires writing out `Z` and `O`. I think formatting standalone timezones for user settings are an important use case.

I would not be opposed to using better names, e.g. `ZoneGeneric` and `ZoneGenericShort`. The current letters don't match CLDR fields anyway, and unlike `YMD`, they refer to a specific field, not a group of fields that will be resolved to actual fields through skeletons.